File: app/src/main/java/com/example/punchpad2/NoteConverter.java
package com.example.punchpad2;

import com.stratonotes.NoteEntity;

public class NoteConverter {

    public static Note fromEntity(NoteEntity entity) {
        Note note = new Note();
        note.id = entity.getId();
        note.content = entity.getContent();
        note.timestamp = entity.getLastEdited();
        note.hidden = entity.isHiddenFromMain();
        note.folderId = entity.getFolderId();
        note.favorited = entity.isFavorite();
        note.large = entity.isLarge();
        note.trashed = entity.isTrashed();
        return note;
    }

    public static NoteEntity toEntity(Note note) {
        return new NoteEntity(
                note.id,
                note.folderId,
                note.content,
                note.timestamp,        // createdAt
                note.timestamp,        // lastEdited
                note.favorited,
                note.hidden,
                note.large,
                note.trashed
        );
    }
}

File: app/src/main/java/com/stratonotes/NoteConverter.kt
package com.stratonotes
@Suppress("unused")
object NoteConverter {
    fun fromEntity(entity: NoteEntity): Note {
        return Note().apply {
            id = entity.id
            folderId = entity.folderId
            content = entity.content
            timestamp = entity.lastEdited
            isHidden = entity.isHiddenFromMain
            isFavorited = entity.isFavorite
            isLarge = entity.isLarge
            isTrashed = entity.isTrashed
        }
    }

    fun toEntity(note: Note): NoteEntity {
        return NoteEntity(
            id = note.id,
            folderId = note.folderId,
            content = note.content ?: "",
            createdAt = note.timestamp,   // Adjust as needed
            lastEdited = note.timestamp,
            isFavorite = note.isFavorited,
            isHiddenFromMain = note.isHidden,
            isLarge = note.isLarge,
            isTrashed = note.isTrashed
        )
    }
}
